pkg/cfn/builder/api_test.go

```diff
@@ -2162,10 +2161,9 @@ var _ = Describe("CloudFormation template builder API", func() {
 
 		roundtrip()
 
-		It("should only have EKS and Fargate resources", func() {
+		It("should only have EKS resources", func() {
```

Comment:
Updating this test is fine, but having another test with a `clusterTemplate` containing Fargate profiles, to test the path adding the Fargate role would be even better! 👍 
For example:
```go
	Context("with Fargate profiles", func() {
		cfg, ng := newClusterConfigAndNodegroup(true)
		name := "test-fargate-profile"
		cfg.Metadata.Name = name
		cfg.FargateProfiles = []*api.FargateProfile{
			{
				Name: "fp-dev",
				Selectors: []api.FargateProfileSelector{
					{Namespace: "dev"},
				},
			},
		}
		build(cfg, "eksctl-"+name+"-cluster", ng)
		roundtrip()

		It("should have the Fargate pod execution role", func() {
			Expect(clusterTemplate.Resources).To(HaveKey("FargatePodExecutionRole"))
		})
	})
```

Reply:
And once this is added, you can even tick the "_Added tests that cover your change (if possible)_" box at the top of the PR! 😉

Reply:
yup it's making sense to have additional test with fargate enabled, let me make an update for the same :100: